FirebaseAdmin/FirebaseAdmin.Tests/Auth/AuthBuilder.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using FirebaseAdmin.Auth.Jwt;
+using FirebaseAdmin.Auth.Multitenancy;
+using FirebaseAdmin.Tests;
+using FirebaseAdmin.Util;
+using Google.Apis.Util;
+
+namespace FirebaseAdmin.Auth.Tests
+{
+    /// <summary>
+    /// An abstraction for creating instances of <see cref="AbstractFirebaseAuth"/> for testing.
+    /// When a tenant ID is configured creates <see cref="TenantAwareFirebaseAuth"/> instances.
+    /// Otherwise creates <see cref="FirebaseAuth"/> instances. Test cases can further customize
+    /// the components of <c>AbstractFirebaseAuth</c> to initialize via <see cref="TestOptions"/>.
+    /// </summary>
+    public sealed class AuthBuilder
+    {
+        internal IClock Clock { get; set; }
+
+        internal RetryOptions RetryOptions { get; set; }
+
+        internal string ProjectId { get; set; }
+
+        internal IPublicKeySource KeySource { get; set; }
+
+        internal string TenantId { get; set; }
+
+        public AbstractFirebaseAuth Build(TestOptions options)
+        {
+            if (this.TenantId != null)
+            {
+                var args = TenantAwareFirebaseAuth.Args.CreateDefault(this.TenantId);
+                this.PopulateArgs(args, options);
+                return new TenantAwareFirebaseAuth(args);
+            }
+            else
```

Comment:
Opinion: The else isn't needed here

Reply:
Wasn't my first choice either. But C# scoping rules do not allow using the same local variable name (args) without the else block. I've renamed the first instance to tenantArgs, and removed the else block.